apps/scratch/src/utilities/scratch-block.ts

```diff
@@ -0,0 +1,80 @@
+import { WorkspaceChangeEvent } from "../types/scratch-workspace";
+import type { Block } from "scratch-blocks";
+
+const minStackSize = 2;
+
+export type DeletedBlockInfo = Array<{ id: string; type: string }>;
```

Comment:
This will be easier to read:
```suggestion
export type DeletedBlockInfo = { id: string; type: string }[];
```

Also I would in general suggest to never define a type as `X[]` for some type X. Instead define the type `X` and use `X[]` wherever you want an array of that type.
Most of the time, you will need to use both and then it is much easier to deal with all use cases if you have the type of the array element available as well.